README.md

```diff
@@ -47,4 +47,43 @@ Options:
       --stream                 Use streaming mode for large files (writes output immediately)
   -h, --help                   Print help
   -V, --version                Print version
+```
+
+## Performance ⚡
+
+csvmd is built for speed and efficiency. Here are some benchmarks showing processing times for various scenarios:
+
+### Real-world Data Processing
+
+| Dataset | Rows | Input Size | Processing Time | Memory Usage |
+| --- | --- | --- | --- | --- |
+| Employee Records | 1,000 | 71KB | 4ms | 3MB |
+| Employee Records | 10,000 | 731KB | 10ms | 8MB |
+| Employee Records | 100,000 | 7MB | 150ms | 66MB |
+| Employee Records | 200,000 | 13MB | 180ms | ~100MB |
+
```

Comment:
Why is memory usage so high in this scenario? Don't change anything unless this is wrong - just explain.